protocol/bc/asset.go

```diff
@@ -12,12 +12,12 @@ import (
 // initial block of the chain where it appears.
 type AssetID [32]byte
 
-func (a AssetID) String() string                { return Hash(a).String() }
-func (a AssetID) MarshalText() ([]byte, error)  { return Hash(a).MarshalText() }
-func (a *AssetID) UnmarshalText(b []byte) error { return (*Hash)(a).UnmarshalText(b) }
-func (a *AssetID) UnmarshalJSON(b []byte) error { return (*Hash)(a).UnmarshalJSON(b) }
-func (a AssetID) Value() (driver.Value, error)  { return Hash(a).Value() }
-func (a *AssetID) Scan(b interface{}) error     { return (*Hash)(a).Scan(b) }
+func (a AssetID) String() string                { return byte32(a).String() }
```

Comment:
I strongly suggest defining AssetID operations in terms of Hash, rather than defining both of them in terms of some third type. Conceptually, an asset ID *is* a hash. Practically, it'll be less code and fewer moving parts that way.

For the purposes of this PR, I am pretty sure these methods can look like

```go
func (a AssetID) String() string                { return NewHash(a).String() }
```

For the future, when AssetID is defined as a protobuf, then we might get rid of all these and instead provide just one or two conversion routines like

```go
func (a AssetID) Hash() Hash { return Hash(a) }
func (a *AssetID) Set(v [32]byte) { (*Hash)(a).Set(v) }
```

something along those lines.